web_group_expand/static/src/js/web_group_expand.js

```diff
@@ -0,0 +1,76 @@
+odoo.define('web_groupby_expand.web_groupby_expand', function (require) {
+"use strict";
+
+var ViewManager = require('web.ViewManager');
+
+ViewManager.include({
+    render_view_control_elements: function (){
+        var res = this._super.apply(this, arguments);
+        if (this.searchview_elements) {
+            var searchview = this.searchview_elements.$searchview_buttons
+            var expand_button = searchview.find('#oe_group_by_expand');
+            var reset_button = searchview.find('#oe_group_by_reset');
+            expand_button.on('click', this.proxy('expand_records'));
+            reset_button.on('click', this.proxy('reset_records'));
+            reset_button.hide();
+            expand_button.hide();
+        }
+        return res;
+    },
+
+    _process_search_data: function () {
+        var res = this._super.apply(this, arguments);
+        if (this.active_view && this.active_view.type == 'list' && this.searchview_elements) {
+            var searchview = this.searchview_elements.$searchview_buttons
+            var expand_button = searchview.find('#oe_group_by_expand');
+            var reset_button = searchview.find('#oe_group_by_reset');
+            var groups = this.active_view.controller.model.localData;
+
+            if (res.groupBy.length > 0) {
+                reset_button.show()
+                expand_button.show()
+            }
+            else {
+                reset_button.hide()
+                expand_button.hide()
+            }
+        }
+        return res;
+    },
+
+    get_search_groups: function (groups) {
+        var current_search_group = {};
+        for (var group in groups) {
+            if (groups[group].count > 0 && groups[group].data.length > 0) {
+                current_search_group[groups[group].id] = groups[group].data;
+            }
+        }
+        return current_search_group;
+    },
+
+    toggle_group_records: function (op, controller) {
+        var current_search_group = this.get_search_groups(controller.model.localData);
+        if (current_search_group) {
+            for (var group in current_search_group) {
+                for (var gp in current_search_group[group]) {
+                    var cur_group = controller.model.localData[current_search_group[group][gp]]
+                    if ((op && !cur_group.isOpen) || (!op && cur_group.isOpen)) {
+                        controller.trigger_up('toggle_group', { group: cur_group })
+                    }
+                }
+            }
+        }
+    },
+    
+    reset_records: function () {
+        var controller = this.active_view.controller;
+        this.toggle_group_records(false, controller)
+    },
+    
+    expand_records: function () {
+        var controller = this.active_view.controller;
+        this.toggle_group_records(true, controller)
+    },
+})
+return ViewManager;
```

Comment:
You don't need to return the ViewManager here. It is already accessible through web.ViewManager

Reply:
Thanks for your review, changes made